app/routes/_auth.tsx

```diff
@@ -15,16 +14,17 @@ export default function AuthRoute() {
     user,
   });
 
-  useEffectNoStrictMode(() => {
-    if (isLoggedIn) {
-      console.debug('Redirecting from auth page to /');
-      navigate('/');
-    }
-  }, [isLoggedIn, navigate]);
-
-  if (loading || isLoggedIn) {
+  if (loading) {
     return <LoadingScreen />;
   }
 
+  if (isLoggedIn) {
+    return (
+      <Redirect to="/" logMessage="Redirecting from auth page to /">
+        <LoadingScreen />
```

Comment:
I don't quite understand why you're wrapping the LoadingScreen. Is it to show Loading while we wait for the useEffect in `Redirect` to run?

Reply:
correct. maybe later we won't want that and will just render null while waiting for it but I didn't want to change the existing behavior in this pr